src/lib/metrics_extractor/lambda_functions_metrics_extractor.py

```diff
@@ -94,6 +96,72 @@ def _data_to_timestream_records(self, lambda_logs: list[LogEntry]) -> list:
         return records, common_attributes
 
     def prepare_metrics_data(self, since_time: datetime) -> tuple[list, dict]:
-        lambda_runs = self._extract_metrics_data(since_time=since_time)
-        records, common_attributes = self._data_to_timestream_records(lambda_runs)
+        self.lambda_runs = self._extract_metrics_data(since_time=since_time)
+        records, common_attributes = self._data_to_timestream_records(self.lambda_runs)
         return records, common_attributes
+
+    ###########################################################################################
+    def generate_event(
+        self,
+        lambdaLogEntry: LogEntry,
+        event_bus_name: str,
+        lambda_aws_account: str,  # region and account where lambda is deployed
+        lambda_aws_region: str,
+    ) -> dict:
+        """
+        Generates json in a form which can be sent to EventBus
+        """
+
+        event_result = (
+            EventResult.FAILURE if lambdaLogEntry.IsErrorEvent else EventResult.SUCCESS
+        )
+        event_state = (
```

Comment:
as discussed - we should not assign SUCCESS status for all logs with "REPORT RequestId:" since each lambda execution (failed or successful) has this log.
Would it be fine if we proceed with the following logic:
event_result = (
            EventResult.FAILURE if lambdaLogEntry.IsErrorEvent else EventResult.INFO
        )
event_state = (
            LambdaManager.LAMBDA_FAILURE_STATE if lambdaLogEntry.IsErrorEvent else None
        )


or should we still assign a Lambda state for all logs with undetermined result, smth like 'EXECUTED' or 'INVOKED'?

Reply:
for state, I guess, we can go with the same approach AWS uses in Glue Workflows - final state is COMPLETED (whatever the result is; only the fact of completion matters)

also I agree with that one:
event_result = (
EventResult.FAILURE if lambdaLogEntry.IsErrorEvent else EventResult.INFO
)

Reply:
sounds good, thank you! updated